unity-environment/Assets/ML-Agents/Scripts/CoreBrainHeuristic.cs

```diff
@@ -44,37 +44,39 @@ public void InitializeCoreBrain(Communicator communicator)
     }
 
     /// Uses the Decision Component to decide that action to take
-    public void DecideAction(Dictionary<Agent, AgentInfo> agentInfo)
+    public void DecideAction(Dictionary<AgentInfo, AgentAction> agentRequest)
     {
         if (coord!=null)
         {
-            coord.GiveBrainInfo(brain, agentInfo);
+            coord.GiveBrainInfo(brain, agentRequest);
         }
 
         if (decision == null)
         {
             throw new UnityAgentsException("The Brain is set to Heuristic, but no decision script attached to it");
         }
 
-        foreach (Agent agent in agentInfo.Keys)
+        foreach (AgentInfo info in agentRequest.Keys)
         {
-            agent.UpdateVectorAction(decision.Decide(
-                agentInfo[agent].stackedVectorObservation,
-                agentInfo[agent].visualObservations,
-                agentInfo[agent].reward,
-                agentInfo[agent].done,
-                agentInfo[agent].memories));
+            agentRequest[info].vectorActions = 
```

Comment:
Just pass "info" here?

Reply:
Depends on what we want to expose in the `Decision.cs` script... It makes sense for us to give the whole `info`, but that puts burden on the developer to look in `.vectorAction`, `.memories`, etc... I am okay with both, we just need to decide. (Similary, Decision. Decide could return an `AgentAction` instead of having one method for `vectorAction` and one for `memories`)